src/webpack.config.ts

```diff
@@ -240,17 +258,28 @@ function webpackConfig(args: Partial<BuildArgs>) {
 					]),
 					new HtmlWebpackPlugin ({
 						inject: true,
-						chunks: [ '../_build/src/main' ],
+						chunks: [ 'src', `../_build/src/main` ],
 						template: 'src/index.html',
-						filename: '../_build/src/index.html'
-					})
-				];
+						filename: `../_build/src/index.html`
+					}),
+					new webpack.optimize.CommonsChunkPlugin({
+						name: 'src',
+						filename: `../_build/src/src.js`,
+						chunks: [`../_build/src/main`, '../_build/tests/unit/all'],
+						minChunks: (module: any) => {
+							if (module.resource && !(/^.*\.(ts)$/).test(module.resource)) {
```

Comment:
what does this do again?

Reply:
So this is the part that actually extracts the source bundle. So basically `src/*/*.ts` will be taken out of `src/main` and `tests/unit/all` and placed in this `src.js` bundle.  The source is not extracted from the functional tests here because we're _assuming_ that the functional tests are only loading things like `.css` files.